documentation/guide-api-first.asciidoc

```diff
@@ -0,0 +1,144 @@
+              <configOptions>
+                <sourceFolder>src/gen/java/main</sourceFolder>
+                <useSwaggerAnnotations>false</useSwaggerAnnotations>
+                <useTags>true</useTags>
+                <interfaceOnly>true</interfaceOnly>
+                <generatePom>false</generatePom>
+              </configOptions>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+    </plugins>
+  </build>
+</profile>
+----
+
+OpenAPI generator project includes support for many languages and frameworks - in our case we picked `jaxrs-spec` server generator.
+Many of the generator templates have further sub-generators, for example for your specific framework. We could, for example use `quarkus` sub-generator to bootstrap a full working Quarkus maven project, but in our case, we want to have more control over the process, therefore we opt for standard `jaxrs` template and pass some additional configuration: 
+
+* `apiPackage` = generated package name for our endpoint interfaces
+* `modelPackage` = generated package name for our API models (DTOs)
+* `sourceFolder` = where do we want to output the generated files, relative to the `output` folder - if we regenerate the files with every build, then it is recommended to put them in `${project.build.directory}/generated-sources/openapi` (default, can be changed via `output` plugin attribute). Final location of our generated Java files will then be `${project.build.directory}/generated-sources/openapi/src/gen/java/main` 
+* `useTags` = group the generated endpoints by schema `tag` attribute
+* `interfaceOnly` = we want to implement the endpoint logic ourselves, so we only want JAX-RS interfaces
+* `generatePom` = we are generating the files into existing project, so we dont want to generate additional maven pom
+
+Let's run the plugin now using `mvn clean compile`. If everything went well, we end up with something like this:
+
+[[img-generated-sources]]
+.Generated sources by OpenAPI generator
+image::images/apifirst-generated-sources.png["Generated sources by OpenAPI generator",scaledwidth="80%",align="center",link="https://devonfw.com/website/pages/docs/images/apifirst-generated-sources.png"]
+
+
+Depending on the configuration of your code generator, you might need to add some additional dependencies to your `pom.xml`, for example for bean validation or Jackson annotations.
+
+The OpenAPI generator has many more configuration options that are outside the scope of this guide. You can find full documentation of the codegen plugin in the https://github.com/OpenAPITools/openapi-generator/tree/master/modules/openapi-generator-maven-plugin[projects GitHub repository].
+
+
+=== Generate once vs generate always
+
+Depending on our needs, we may either want to generate the interfaces and models once and afterwards copy them to our project as general source code files, or treat them as immutable generated assets, that we generate anew with every build.
```

Comment:
... or treat them as immutable generated assets(no comma) that we generate anew with every build.